src/main/scala/sha3.scala

```diff
@@ -74,69 +93,83 @@ class Sha3AccelImp(outer: Sha3Accel)(implicit p: Parameters) extends LazyRoCCMod
   io.resp.valid := Bool(false) //Sha3 never returns values with the resp
   //mem
   //busy
-
-  val ctrl = Module(new CtrlModule(W,S)(p))
-
-  ctrl.io.rocc_req_val   <> io.cmd.valid
-  io.cmd.ready := ctrl.io.rocc_req_rdy
-  ctrl.io.rocc_funct     <> io.cmd.bits.inst.funct
-  ctrl.io.rocc_rs1       <> io.cmd.bits.rs1
-  ctrl.io.rocc_rs2       <> io.cmd.bits.rs2
-  ctrl.io.rocc_rd        <> io.cmd.bits.inst.rd
-  io.busy := ctrl.io.busy
-
-  val dmem_data = Wire(Bits())
-  private def dmem_ctrl(req: DecoupledIO[HellaCacheReq]) {
-    req.valid := ctrl.io.dmem_req_val
-    ctrl.io.dmem_req_rdy := req.ready
-    req.bits.tag := ctrl.io.dmem_req_tag
-    req.bits.addr := ctrl.io.dmem_req_addr
-    req.bits.cmd := ctrl.io.dmem_req_cmd
-    req.bits.size := ctrl.io.dmem_req_size
-    req.bits.data := dmem_data
-    req.bits.signed := Bool(false)
-    req.bits.phys := Bool(false)
-  }
-
-  outer.dmemOpt match {
-    case Some(m) => {
-      val dmem = m.module
-      dmem_ctrl(dmem.io.req)
-      io.mem.req <> dmem.io.mem
-      io.ptw.head <> dmem.io.ptw
-
-      dmem.io.status.valid := io.cmd.fire()
-      dmem.io.status.bits := io.cmd.bits.status
-      dmem.io.sfence := ctrl.io.sfence
+  if (p(Sha3BlackBox)) {
+    require(!p(Sha3TLB).isDefined, "Blackbox SHA3 does not support Dmemmodule")
+
+    val sha3bb = Module(new Sha3BlackBox(0))
```

Comment:
Its strange this was parameterized at all given its hard coded to 0.

Reply:
fixed